test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -940,6 +942,41 @@ public void waitNoPendingTasksOnAll() throws Exception {
         assertNoTimeout(clusterAdmin().prepareHealth(TEST_REQUEST_TIMEOUT).setWaitForEvents(Priority.LANGUID).get());
     }
 
+    /**
+     * Waits for the node {@code viaNode} to see {@code masterNodeName} as the master node in the cluster state and asserts that.
+     * Note that this does not guarantee that all other nodes in the cluster are on the same cluster state version already.
+     *
+     * @param viaNode the node to check the cluster state one
+     * @param masterNodeName the master node name that we wait for
+     */
+    public void awaitAndAssertMasterNode(String viaNode, String masterNodeName) throws Exception {
+        awaitClusterState(
+            logger,
+            viaNode,
+            state -> Optional.ofNullable(state.nodes().getMasterNode()).map(m -> m.getName().equals(masterNodeName)).orElse(false)
+        );
+        var nodes = client(viaNode).admin().cluster().prepareState(TEST_REQUEST_TIMEOUT).setLocal(true).get().getState().nodes();
```

Comment:
These API calls (also the ones below) have `setLocal(true)` because that's how they will be run once the API runs on the local node in the future.